Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,429 @@
+     */
+    final function __destruct()
+    {
+        try {
+            $this->manager->disconnect();
+        } catch (\Exception $connectionException) {
+            $this->getExceptionHandler()($connectionException);
+        }
+    }
+
+    /**
+     * @param $host
+     * @param string $username
+     * @param string $password
+     * @param string $format
+     */
+    public function configure($host, $username, $password, $format = self::FORMAT_JSON)
+    {
+        $this->format = $format;
+    }
+
+    /**
+     * @throws \AMQPException
+     */
+    public function connect()
+    {
+        $this->manager->connect();
+    }
+
+    /**
+     * @throws \AMQPException
+     */
+    public function disconnect()
+    {
+        $this->manager->disconnect();
+    }
+
+    /**
+     * @return QueueMessage|QueueMessageInterface
+     */
+    public function createQueueMessage($queueName = null, $options = [])
+    {
+        $msg = new QueueMessage();
+        $msg->setContext(new Context());
+        return $msg;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isConnected()
+    {
+        return $this->manager->isConnected();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isSubscribed()
+    {
+        return null !== $this->queue;
+    }
+
+    /**
+     * @todo verify the need of this function
+     * @param string $queue
+     */
+    public function subscribe($queue)
+    {
+        /*if ($this->currentEnvelope) {
+            throw new \RuntimeException('Don\'t be greedy and process the message you already have!');
+        }
+        $this->queue = $this->manager->getQueue((string) $queue);*/
+    }
+
+    /**
+     * @todo verify the need of this function
+     * @throws \Exception
+     */
+    public function unSubscribe()
+    {
+        /*$this->disconnect();
+        $this->queue = $this->currentEnvelope = null;*/
+    }
+
+    /**
+     * @todo verify the need of this function
+     * {@inheritdoc}
+     */
+    public function ack()
+    {
+        /*if (!$this->currentEnvelope) {
+            throw new \RuntimeException('You must first receive a message, before acking it');
+        }
+
+        $this->queue->ack($this->currentEnvelope->getDeliveryTag());
+        $this->currentEnvelope = null;*/
+    }
+
+    /**
+     * @todo verify the need of this function
+     * {@inheritdoc}
+     */
+    public function nack()
+    {
+        /*if (!$this->currentEnvelope) {
+            throw new \RuntimeException('You must first receive a message, before nacking it');
+        }
+
+        $this->queue->nack($this->currentEnvelope->getDeliveryTag(), AMQP_REQUEUE);
+        $this->currentEnvelope = null;*/
+    }
+
+    /**
+     * @todo verify the need of this function
+     * {@inheritdoc}
+     */
+    public function doDestroy()
+    {
+        try {
+            $this->manager->disconnect();
+        } catch (\Exception $e) {
+            $this->getExceptionHandler()($e);
```

Comment:
The same comment I added before in the `__destruct()` method applies here

Reply:
It was modified.